cirq-google/cirq_google/workflow/qubit_placement.py

```diff
@@ -73,3 +81,130 @@ def _json_dict_(self) -> Dict[str, Any]:
 
     def __repr__(self) -> str:
         return _compat.dataclass_repr(self, namespace='cirq_google')
+
+
+def default_topo_node_to_qubit(node: Any) -> cirq.Qid:
+    """The default mapping from `cirq.NamedTopology` nodes and `cirq.Qid`.
+
+    There is a correspondence between nodes and the "abstract" Qids
+    used to construct un-placed circuit. `cirq.get_placements` returns a dictionary
+    mapping from node to Qid. We use this function to transform it into a mapping
+    from "abstract" Qid to device Qid. This function encodes the default behavior used by
+    `RandomDevicePlacer`.
+
+    If nodes are tuples of integers, map to `cirq.GridQubit`. Otherwise, try
+    to map to `cirq.LineQubit` and rely on its validation.
+
+    Args:
+        node: A node from a `cirq.NamedTopology` graph.
+
+    Returns:
+        A `cirq.Qid` appropriate for the node type.
+    """
+
+    try:
+        return cirq.GridQubit(*node)
+    except TypeError:
+        return cirq.LineQubit(node)
+
+
+@lru_cache()
+def _cached_get_placements(
+    problem_topo: 'cirq.NamedTopology', device: 'cirq.Device'
+) -> List[Dict[Any, 'cirq.Qid']]:
+    """Cache `cirq.get_placements` onto the specific device."""
+    return get_placements(
+        big_graph=_Device_dot_get_nx_graph(device), small_graph=problem_topo.graph
+    )
+
+
+def _get_random_placement(
+    problem_topology: 'cirq.NamedTopology',
+    device: 'cirq.Device',
+    rs: np.random.RandomState,
+    topo_node_to_qubit_func: Callable[[Any], 'cirq.Qid'] = default_topo_node_to_qubit,
+) -> Dict['cirq.Qid', 'cirq.Qid']:
+    """Place `problem_topology` randomly onto a device.
+
+    This is a helper function used by `RandomDevicePlacer.place_circuit`.
+    """
+    placements = _cached_get_placements(problem_topology, device)
+    if len(placements) == 0:
+        raise CouldNotPlaceError
+    random_i = int(rs.random_integers(0, len(placements) - 1, size=1))
```

Comment:
`random_integers` is deprecated, use `randint` instead. 

```suggestion
    random_i = rs.randint(len(placements))
```